gnocchi/cli.py

```diff
@@ -86,7 +86,11 @@ def change_sack_size():
     ])
     conf = service.prepare_service(conf=conf)
     s = storage.get_incoming_driver(conf.incoming)
-    report = s.measures_report(details=False)
+    try:
+        report = s.measures_report(details=False)
+    except Exception:
+        # issue is already logged by measures_report
```

Comment:
On https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/storage/incoming/_carbonara.py#L49

Change by `raise NumSackReadingError(e)` (or any better name, sorry)

and in this patch, line 91 just change `except Exception` by `except NumSackReadingError`. That way you're sure you're only ignoring this for a good reason.

WDYT?